spark/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -119,28 +166,35 @@ case class CheckpointInstance(
    *    Single part checkpoint.
    * 3. For Multi-part [[CheckpointInstance]]s corresponding to same version, the one with more
    *    parts is greater than the one with less parts.
+   *  4. For V2 Checkpoints corresponding to same version, we use the fileName as tie breaker.
```

Comment:
What is the semantics of using file name to break ties? Put another way, when would that make a difference? Reaching step 4 of the comparison means that we're dealing with the same version, same format, and same number of parts. What else could differ in the name?

Reply:
Update: .json vs. .parquet... tho I'm not sure we actually want to prefer json merely because it lexically precedes parquet?

Reply:
oh, uuid-named... right